onnxscript/rewriter/onnxruntime/bfloat16_utils/bfloat16_converter.py

```diff
@@ -57,6 +57,12 @@ def _insert_cast_nodes_for_bfloat16_to_float16_to_outputs(node: ir.Node, index:
     for idx, graph_or_function_output in enumerate(node.graph.outputs):
         if graph_or_function_output == node.outputs[index]:
             node.graph.outputs[idx] = cast.outputs[0]
+    # Swap the output name of the node with the output name of the cast node to
+    # preserve the output name in the graph
+    node.outputs[index].name, cast.outputs[0].name = (
```

Comment:
One caveat is if the incoming graph is produced by IR and contains names like val_0; we would have a name collision since we don't currently ensure name uniqueness in the IR, and the new output's name will again be val_0; so _maybe_ adding the _bfloat16 suffix can be a better solution for now; the alternative is to create a pass to reconcile name collisions; or improve the naming authority to produce unique names

Reply:
Maybe I should use graph.insert_before/after? Those should reasonably cover the same naming authority?

Reply:
prepend and append calls the graphs methods internally so they should be the exact same.

Reply:
I see. I thought naming authority lived inside the graph, and managed them not to collide.

Reply:
> I see. I thought naming authority lived inside the graph

this is true

> and managed them not to collide.

This is not implemented yet